flake-module.nix

```diff
@@ -133,6 +153,34 @@ in
                   This is an internal option, not meant to be set by the user.
                 '';
               };
+
+              # Derived options
+
+              finalPackages = mkOption {
+                type = types.attrsOf raw;
+                readOnly = true;
+                default = config.haskellPackages.extend config.finalOverlay;
+                defaultText = lib.literalMD "internal definition";
+                description = ''
+                  The final package set, based `haskellPackages` plus
+                  the additions and overrides specified in the other options.
+                '';
+              };
+              finalOverlay = mkOption {
+                type = types.raw;
+                readOnly = true;
+                internal = true;
+                default = lib.composeManyExtensions [
```

Comment:
@roberth Why do `finalPackages` and `finalOverlay` use `default` (in `options`) to define their values, unlike [`outputs`](https://github.com/srid/haskell-flake/pull/63/files#diff-0549e7a291dba4b0c8148bb5c499a14bb7c1cd0aa61e74e31623467004ad2b6fR149) which uses `config` to do it? What is the difference between the two, in general?

Reply:
The difference is the `mkOverride` priority that is assigned to the definition. In case of a `readOnly = true` option, it doesn't really matter which one you use, because only one definition is allowed anyway, making override priorities irrelevant.

I think placing definitions in `config` is a good habit though, so I'll change this.

Reply:
> I think placing definitions in `config` is a good habit though, so I'll change this.

Cool; it would also minimize the diff in this PR, I imagine, since all `config` stuff lives in `haskell-project.nix`.

Reply:
To move it back, I'd have to change `haskell-project.nix` to a module instead of a function to a `config` value. Basically this 
 - https://github.com/srid/haskell-flake/pull/63/files#r1100250488

I guess why not